src/helix-ui/elements/HXModalElement.js

```diff
@@ -37,50 +37,31 @@ export class HXModalElement extends HXElement {
         return `<style>${shadowStyles}</style>${shadowMarkup}`;
     }
 
-    static get observedAttributes () {
+    static get $observedAttributes () {
         return [ 'open' ];
     }
 
-    constructor () {
-        super();
-        this._close = this._close.bind(this);
-        this._keyUp = this._keyUp.bind(this);
+    $onCreate () {
+        this._onBtnClose = this._onBtnClose.bind(this);
+        this._onKeyUp = this._onKeyUp.bind(this);
     }
 
-    connectedCallback () {
+    $onConnect () {
         this.$upgradeProperty('open');
         this.setAttribute('aria-hidden', !this.open);
-
-        this._btnClose.addEventListener('click', this._close);
-        document.addEventListener('keyup', this._keyUp);
```

Comment:
The `keyup` event listener should only live as long as the modal is open.  (Pressing ESC when the modal is closed would result in no change to its state.)

I've refactored the logic so that the listener is added only when the modal is opened.